src/SoftRobots/component/constraint/model/CableModel.inl

```diff
@@ -261,6 +321,241 @@ void CableModel<DataTypes>::checkIndicesRegardingState()
+            radii[i] = radius;
+        msg_warning(this) << "Size of centers and radii list not the same. Will apply first radius to all the effect areas.";
+    }
+    
+    computePointsActionArea();
+}
+
+
+template<class DataTypes>
+void CableModel<DataTypes>::computePointsActionArea()
+{
+    // Implementing continuous Dijkstra as a geodesic distance measure
+    const unsigned int id_method = d_method.getValue().getSelectedId();
+    unsigned int nbCenters = d_indices.getValue().size();      
+    const SetIndexArray &indices = d_indices.getValue();
+
+    ReadAccessor<Data<VecCoord>> cablePositions = m_state->readPositions();
+    ReadAccessor<Data<VecCoord>> centers = d_centers;    
+    if (m_hasCenters)
+    {
+        m_alpha.clear();
+        m_alpha.resize(nbCenters);
+        m_beta.clear();
+        m_beta.resize(nbCenters);
+        m_closestTriangle.clear();
+        m_closestTriangle.resize(nbCenters);
+    } 
+    ReadAccessor<Data<vector<Real>>> maxDistance = d_radii;
+    vector<Real> m_totalRatio;
+    m_areaIndices.clear();
+    m_areaIndices.resize(nbCenters);
+    m_ratios.clear();
+    m_ratios.resize(nbCenters);
+    m_totalRatio.clear();
+    m_totalRatio.resize(nbCenters);
+    
+
+    // Iterate for each cable attachment
+    typedef pair<Real,BaseMeshTopology::PointID> DistanceToPoint;
+    for (unsigned int i=0; i<nbCenters; i++)
+    {   
+        // Init
+        set<DistanceToPoint> queue; 
+        typename set<DistanceToPoint>::iterator qit;
+        vector<Real> distances(l_surfaceTopology.get()->getNbPoints(),maxDistance[i]);
+
+        // If cable attachment does not match with a mesh vertice
+        // Project on closest triangle and distribute distance to its vertices 
+        if (m_hasCenters)
+        {
+            const TrianglesAroundVertex& trianglesAroundClosestVertex = l_surfaceTopology.get()->getTrianglesAroundVertex(indices[i]); // Triangles connected to closest vertice
+            Coord projectionOnTriangle;
+            Real minDistanceToTriangle = std::numeric_limits<Real>::max(); unsigned int closestTriangleId = 0; Coord closestProjectionOnTriangle;
+            for(unsigned int j=0; j<trianglesAroundClosestVertex.size(); j++)
+            {
+                const Triangle triangle =  l_surfaceTopology.get()->getTriangle(trianglesAroundClosestVertex[j]);
+                const Real distanceToTriangle = getDistanceToTriangle(centers[i], triangle, projectionOnTriangle);
+
+                if(distanceToTriangle < minDistanceToTriangle)
+                {
+                    minDistanceToTriangle = distanceToTriangle;
+                    closestTriangleId = j;
+                    closestProjectionOnTriangle = projectionOnTriangle;
+                }
+            }
+
+            const Triangle closestTriangle =  l_surfaceTopology.get()->getTriangle(trianglesAroundClosestVertex[closestTriangleId]);
+
+            
+            for(unsigned int j=0; j<closestTriangle.size(); j++)
```

Comment:
In the future, prefer range for loops when possible
https://en.cppreference.com/w/cpp/language/range-for